shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -312,11 +318,21 @@ public final WindowInsets onApplyWindowInsets(@NonNull WindowInsets insets) {
     viewportMetrics.viewInsetBottom = insets.getSystemWindowInsetBottom();
     viewportMetrics.viewInsetLeft = 0;
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
```

Comment:
Do you think it be worth it to still add a test for the non Q case in this PR where these are set to zero? It would just basically be verifying the current behavior and making sure that these lines were guarded and didn't cause problems on lower SDK versions. So some but not a huge amount of value from them.

Reply:
I don't know the root cause of the problem, but I believe that because the TargetApi is set to 20 and the tests only use up to SDK 16 at runtime, the `onApplyWindowInsets` method gets completely ignored by the test framework (I tried putting print statements in the method and they weren't being called). So, event his might be blocked on upgrading Robolectric :\

Reply:
Ah, right. The entire method here requires at least API 20, and our setup only includes API 16 right now.

If you wanted to test just the non-Q parts of this method, you should be able to bring in API 21 without any trouble. Robolectric has supported it since 3.0 (we're on 3.8). It's 29/Q specifically that required upgrading to 4.3 and is currently blocked. But doing this has some extra overhead where it would require bringing in another robolectric android package for the right SDK [like this one](https://repo1.maven.org/maven2/org/robolectric/android-all/5.0.0_r2-robolectric-0/) and adding it to CIPD as described [here](https://github.com/flutter/engine/blob/master/shell/platform/android/test/README.md#updating-a-cipd-dependency). So at that point I'm not sure if it's worth the hassle. There may be some unknown unknowns that cause it to take extra time.